src/RegistrationMsg.chpl

```diff
@@ -96,6 +99,199 @@ module RegistrationMsg
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        var lens = st.attrib("%s_lengths".format(name));
+        if (lens.startsWith("Error:")) { 
+            var errorMsg = lens;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        repMsg = "segarray+created %s+created %s+created %s".format(segs, vals, lens);
+
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+
+    /* 
+    Parse, execute, and respond to a generic attach message
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg payload: request containing (dtype+name)
+    :type payload: string
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc genAttachMsg(cmd: string, payload: string, 
+                                            st: borrowed SymTab): MsgTuple throws {
+        var repMsg: string; // response message
+
+        var ele_parts = payload.split("+");
+        var dtype = ele_parts[0];
+        var name = ele_parts[1];
+
+        if dtype == "infer" {
+            // Try to determine the type from the entries in the symbol table
+            // Should try __RegisteredSymbols__ instead
+            var symbols = st.info("__RegisteredSymbols__");
+            regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "Attempting to find type of registered element '%s'".format(name));
+
+            if st.contains(name) {
+                // Easy case where full name given matches an entry, pdarray or Strings
+                dtype = "simple";
+            } else if st.contains("%s.categories".format(name)) && st.contains("%s.codes".format(name)) {
+                dtype = "categorical";
+            } else if st.contains("%s_segments".format(name)) && st.contains("%s_values".format(name)) {
+                // Important to note that categorical has a .segments while segarray uses _segments
+                dtype = "segarray";
+            } else {
+                throw getErrorWithContext(
+                                    msg="Unable to determine type for given name: %s".format(name),
+                                    lineNumber=getLineNumber(),
+                                    routineName=getRoutineName(),
+                                    moduleName=getModuleName(),
+                                    errorClass="ValueError"
+                                    );
+            }
+        }
+
+        // type possibilities for pdarray and strings
+        var simpleTypes: list(string) = ["pdarray","int64", "uint8", "uint64", "float64", "bool", "strings", "string", "str"];
+        if simpleTypes.contains(dtype.toLower()) {
+            dtype = "simple";
+        }
```

Comment:
I don't think this is necessary because we should have already overrode the dtype above?

In this if:
```Chapel
            if st.contains(name) {
                // Easy case where full name given matches an entry, pdarray or Strings
                dtype = "simple";
```

Reply:
Not exactly. If the dtype given is infer, then it can hit that `if` where the dtype is set to `"simple"` only if the name matches a name in the table.

However, if a user passes in the result from `my_pdarray.dtype` then we have to handle `int64`. This is checking all possible supported types that could be provided from the user and simplifying them to a single case to be used in the `select` block

Reply:
ahh i see!